standard/attributes.md

```diff
@@ -858,6 +860,277 @@ For invocations that occur within field or event initializers, the member name u
+
+#### §The-DoesNotReturn-Attribute The DoesNotReturn attribute
+
+Specifies that a given method never returns.
+
+> *Example*: Consider the following:
+>
+> <!-- Example: {template:"standalone-lib", name:"DoesNotReturnAttribute"} -->
+> ```csharp
+> public class X
+> {
+>     [DoesNotReturn]
+>     private void FailFast() =>
+>         throw new InvalidOperationException();
+>
+>     public void SetState(object? containedField)
+>     {
+>         if ((!isInitialized) || (containedField == null))
+>         {
+>             FailFast();
+>         }
+>         // null check not needed.
+>         _field = containedField;
+>     }
+> 
+>     private bool isInitialized = false;
+>     private object _field;
+> }
+> ```
+>
+> The presence of the attribute helps the compiler in a number of ways. First, the compiler can issue a warning if there's a path where the method can exit without throwing an exception. Second, the compiler can suppress nullable warnings in any code after a call to that method, until an appropriate catch clause is found. Third, the unreachable code won't affect any null states.
+>
+> The compiler can't change reachability (§13.2) or definite assignment (§9.4) analysis based on the presence of this attribute. A compiler can use this attribute only to impact nullability warnings. *end example*
+
+#### §The-DoesNotReturnIf-Attribute The DoesNotReturnIf attribute
+
+Specifies that a given method never returns if the associated `bool` parameter has the specified value.
+
+> *Example*: Consider the following:
+>
+> <!-- Example: {template:"standalone-lib", name:"DoesNotReturnIfAttribute", expectedWarnings:["CS0414"]}  -->
+> ```csharp
+> #nullable enable
+> public class X
+> {
+>     private void ThrowIfNull([DoesNotReturnIf(true)] bool isNull, string argumentName)
+>     {
+>         if (!isNull)
+>         {
+>             throw new ArgumentException(argumentName, $"argument {argumentName} can't be null");
+>         }
+>     }
+>
+>     public void SetFieldState(object containedField)
+>     {
+>         ThrowIfNull(containedField == null, nameof(containedField));
+>         // unreachable code when "isInitialized" is false:
+>         _field = containedField;
+>     }
+> 
+>     private bool isInitialized = false;
+>     private object _field = default!;
+> }
+> ```
+>
+> *end example*
+
+#### §The-MaybeNull-Attribute The MaybeNull attribute
+
+Specifies that a non-nullable return value may be null.
+
+> *Example*: Consider the following generic method:
+>
+> <!-- Example: {template:"code-in-class-lib-without-using", name:"MaybeNull1Attribute", replaceEllipsis:true, customEllipsisReplacements: ["return default;"]} -->
+> ```csharp
+> #nullable enable
+> public T? Find<T>(IEnumerable<T> sequence, Func<T, bool> predicate) { ... }
```

Comment:
Should this be valid C# 8? `T?` is only allowed in C# 9+ when there is no class or struct constraint.

Reply:
The suggestion for this is to use the same declaration as line 1026, but explain that `Find` could return a `null` value.